WordPress/gutenberg
# [Mobile] Update the bottom sheet header

## Description
This PR Refactors the Navigation Header Component so that it is more compostable, easier to understand and updates the usage of the component. 

## How has this been tested?
1. Load up this PR on a mobile simulator.
2. Navigate to different Bottoms Sheets. Notice that they look the same as expected. 
Bottoms sheets to test out. 

- Button Block. 
  - Button Width.
  - Link Settings.
- Image block
  - Alt text
  - Size Picker
- Cover Block.
  - Edit Focal Point
  - Color Picker
- Columns Block
  - Columns Selector
- Paragraph Block
  - Font Site Settings. 
  - Colour Settings
  - Gradient Settings
- Help Section. 


## Screenshots 
The Bottom Sheet Header should look exactly the same as before. 
Something like this.
<img width="432" alt="Screen Shot 2021-08-25 at 3 02 27 PM" src="https://user-images.githubusercontent.com/115071/130870397-dbeee8ca-f28f-4fb1-ae0d-0cd74d64cec6.png">


## Types of changes
Refactor

## Checklist:
